notify/impl.go

```diff
@@ -474,17 +488,41 @@ func (n *PagerDuty) Notify(ctx context.Context, as ...*types.Alert) (bool, error
 		details[k] = tmpl(v)
 	}
 
-	msg := &pagerDutyMessage{
-		ServiceKey:  tmpl(string(n.conf.ServiceKey)),
-		EventType:   eventType,
-		IncidentKey: hashKey(key),
-		Description: tmpl(n.conf.Description),
-		Details:     details,
+	msg := &pagerDutyMessage{}
+
+	if n.conf.ServiceKey != "" {
+		msg.ServiceKey = tmpl(string(n.conf.ServiceKey))
+		msg.EventType = eventType
+		msg.IncidentKey = hashKey(key)
+		msg.Description = tmpl(n.conf.Description)
+		msg.Details = details
+		n.conf.URL = "https://events.pagerduty.com/generic/2010-04-15/create_event.json"
+	} else {
+		msg.RoutingKey = tmpl(string(n.conf.RoutingKey))
```

Comment:
RoutingKey is an integration key provided by pagerduty, should not be processed like a template.

Reply:
Strangely, ServiceKey (which is the v1 equivalent) was being processed as a template as well in the past